### ISSUE TITLE
Key vault references supported in Azure function (python) on free linux app service?

### ISSUE THREAD

**Issue by rafaelschlatter** [Azure/azure-functions-host#5346]
Hi,

I have an Azure function (python) app, which is running on a free linux app service plan. I am using `FUNCTION_EXTENSION_VERSION` = ~2. I want to integrate this function app with a key vault and reference secrets in the app configuration.

I have enabled a system-assigned identity for the function app, and created a key-vault access policy for the function app, with `LIST` and `GET` permissions for secrets. I reference the key vault secret in the app config like this:

@Microsoft.KeyVault(SecretUri=secret-identifier)

The key vault reference is being displayed as resolved in the app config, and Azure function diagnostics > Availability and performance > Function app down or reporting errors > key vault application settings diagnostics shows everything okay.

When testing the function app, it is not able to retrieve the secret. The key vault does not have any network restrictions. Does this have something to do with the free app service plan?

I found this: https://azure.microsoft.com/en-in/updates/general-availability-of-key-vault-references-in-app-service-and-azure-functions/, which does not explicitly state the free app service plan.

Any hints on how to get this to work?

Regards,
Rafael

**Comment by balag0** [comment]
Is it possible for you to share the app name?

How are you reading the appsetting in the app? Are you getting an empty value or the unresolved reference to secret?

**Comment by rafaelschlatter** [comment]
Hi balag0,

Thanks for your response. I managed to resolve this, however I do not know what the problem was initially.

I reconfigured the app including the key vault reference in the app settings of the function app, and it started working. Suggests a spelling mistake in the app settings, which would be a bit embarrassing.

To your question, I was reading all app settings with `os.environ["app-settings-name"]`. The error I was getting was "The function runtime is unable to start", so did not even reach the code execution.

I apologize. This can be closed.

Regards,
Rafael
